lib/hash_digest.rb

```diff
@@ -0,0 +1,29 @@
+require 'digest'
+
+class HashDigest
+  def self.generate(hash = {}, query_field = :tags)
+    raise ArgumentError.new("query_field must be `:tags` or `:links`") unless %i{tags links}.include?(query_field)
+
+    return nil if hash.empty?
+
+    wrapped = {}
+    wrapped[query_field] = sort_hash(hash)
+    Digest::MD5.new.update(wrapped.to_s).hexdigest
```

Comment:
I'd imagine it'd be a bit safer to use a SHA hash to just reduce the fatal risk of collisions further.